### PR TITLE
Patched 2 vulnerabilities, XSS and SQL Injection

### PR DESCRIPTION
**Pull request by 0x43434343**

### CODE REVIEW

File: inc/admin-subpage.php [hunk 1]
```diff
@@ -458,7 +461,10 @@ protected function bulk_actions( $which = '' ) {
         submit_button( __( 'Apply', 'contact-form-cfdb7' ), 'action', '', false, array( 'id' => "doaction$two" ) );
         echo "\n";
         $nonce = wp_create_nonce( 'dnonce' );
-        echo "<a href='".$_SERVER['REQUEST_URI']."&csv=true&nonce=".$nonce."' style='float:right; margin:0;' class='button'>";
+         // Cross Site Scripting ( XSS ) 
+         // filter $_SERVER request will prevent from XSS attack 
+        // esc_html(), esc_attr(), esc_url() <= 
+        echo "<a href='".esc_html($_SERVER['REQUEST_URI'])."&csv=true&nonce=".$nonce."' style='float:right; margin:0;' class='button'>";
```
Comment (arshidkv12) [inline inc/admin-subpage.php:467]: esc_html

File: inc/admin-subpage.php [hunk 2]
```diff
@@ -308,8 +308,10 @@ public function process_bulk_action(){
             $form_ids = esc_sql( $_POST['contact_form'] );
 
             foreach ($form_ids as $form_id):
-
-                $results       = $cfdb->get_results( "SELECT * FROM $table_name WHERE form_id = $form_id LIMIT 1", OBJECT );
+                // first 
+                //form_ids -> SQL Injection 
+                // to fix this issue -> '' will be add
+                $results       = $cfdb->get_results( "SELECT * FROM $table_name WHERE form_id = '$form_id' LIMIT 1", OBJECT );
```
Comment (arshidkv12) [inline inc/admin-subpage.php:314]: form_id

File: inc/admin-subpage.php [hunk 3]
```diff
@@ -352,7 +355,7 @@ public function process_bulk_action(){
 
             $form_ids = esc_sql( $_POST['contact_form'] );
             foreach ($form_ids as $form_id):
-
+               //form_ids -> Sql injection 
                 $results       = $cfdb->get_results( "SELECT * FROM $table_name WHERE form_id = '$form_id' LIMIT 1", OBJECT );
```
Comment (arshidkv12) [inline inc/admin-subpage.php:359]: form_id

**Review by arshidkv12 — COMMENTED** [review]
quote

**Comment by 0x43434343** [comment]
Thanks Arshid , can you request 2 CVE for that one for XSS and the second
for Sql Injection :D ?

**Pull request merged**